tests/sources/xml/test_ead.py

```diff
@@ -739,6 +730,282 @@ def test_get_content_type_transforms_correctly_if_fields_missing():
+                </persname>
+            </origination>
+            """
+        ),
+        parent_element="did",
+    )
+    assert Ead.get_contributors(source_record) == [
+        timdex.Contributor(value="Author, Best E. ( Best Ever )", kind="Creator")
+    ]
+
+
+def test_get_contributors_transforms_correctly_if_fields_blank():
+    source_record = create_ead_source_record_stub(
+        metadata_insert=(
+            """
+            <origination></origination>
+            """
+        ),
+        parent_element="did",
+    )
+    assert Ead.get_contributors(source_record) is None
+
+
+def test_get_contributors_transforms_correctly_if_fields_missing():
+    source_record = create_ead_source_record_stub(parent_element="did")
+    assert Ead.get_contributors(source_record) is None
+
+
+def test_get_contributors_transforms_correctly_if_multiple_contributors():
+    source_record = create_ead_source_record_stub(
+        metadata_insert=(
+            """
+            <origination label="Creator">
+                <persname>
+                    Author, Best E.
+                    <part>( <emph> Best <emph>Ever</emph> </emph> )</part>
+                </persname>
+                <persname>
+                    Author, Better
+                </persname>
+            """
+        ),
+        parent_element="did",
+    )
+    assert Ead.get_contributors(source_record) == [
+        timdex.Contributor(
+            value="Author, Best E. ( Best Ever )",
+            kind="Creator",
+        ),
+        timdex.Contributor(
+            value="Author, Better",
+            kind="Creator",
+        ),
+    ]
+
+
+def test_get_contributors_transforms_correctly_with_source_based_identifiers():
+    source_record = create_ead_source_record_stub(
+        metadata_insert=(
+            """
+            <origination label="Creator">
+                <persname authfilenumber="a001" source="naf">
+                    Author, Best E.
+                    <part>( <emph> Best <emph>Ever</emph> </emph> )</part>
+                </persname>
+                <persname authfilenumber="b001" source="viaf">
+                    Author, Better
+                </persname>
+            </origination>
+            <origination>
+                <famname authfilenumber="c001" source="snac">Fambam</famname>
+            </famname>
+            """
+        ),
+        parent_element="did",
+    )
+    assert Ead.get_contributors(source_record) == [
+        timdex.Contributor(
+            value="Author, Best E. ( Best Ever )",
+            identifier=["https://lccn.loc.gov/a001"],
+            kind="Creator",
+        ),
+        timdex.Contributor(
+            value="Author, Better",
+            identifier=["http://viaf.org/viaf/b001"],
+            kind="Creator",
+        ),
+        timdex.Contributor(
+            value="Fambam", identifier=["https://snaccooperative.org/view/c001"]
+        ),
+    ]
+
+
+def test_get_dates_success():
```

Comment:
I would add a non-range date value as well so we're testing everything

Reply:
Change has been made in commit https://github.com/MITLibraries/transmogrifier/pull/187/commits/95b6b4758db82a5212545b2faa70b3acf0a1e73f)